main.go

```diff
@@ -164,10 +203,117 @@ func main() {
 		setupLog.Info("starting http server", "address", address)
 		log.Fatal(http.ListenAndServe(address, h))
 	}()
+
+	var mutex *sync.Mutex
+
+	k8sClient, err := getK8sClientSet()
+	if err != nil {
+		setupLog.Error(err, "Cannot get the k8s client set")
+		os.Exit(1)
+	}
+
+	stop := make(chan struct{})
+	defer close(stop)
+
+	operatorNamespace, err := getOperatorNamespace(Config.LocalRun)
+	if err != nil {
+		setupLog.Error(err, "Cannot get the operator namespace")
+		os.Exit(1)
+	}
+	setupLog.V(1).Info("operator namespace found", "operatorNamespace", operatorNamespace)
+
+	configMapName, err := getConfigMapName(k8sClient, operatorNamespace, Config.LocalRun)
+	if err != nil {
+		setupLog.Error(err, "operator namespace", operatorNamespace)
+		os.Exit(1)
+	}
+
+	currentConfigMap, err := k8sClient.CoreV1().ConfigMaps(operatorNamespace).Get(context.TODO(), configMapName, metav1.GetOptions{})
+	if err != nil {
+		setupLog.Error(err, "Unable to get ConfigMap", "namespace", operatorNamespace)
+		os.Exit(1)
+	}
+	setupLog.V(1).Info("operator configmap found", "operatorNamespace", operatorNamespace, "configmap name", configMapName)
+
+	mutex = &sync.Mutex{}
```

Comment:
why is this mutex needed ? Seems it can be defined from `WatchForChanges` and doesn't have to be passed to it.